src/backend/distributed/utils/connection_cache.c

```diff
@@ -60,56 +51,20 @@ static void ReportRemoteError(PGconn *connection, PGresult *result, bool raiseEr
 PGconn *
 GetOrEstablishConnection(char *nodeName, int32 nodePort)
 {
+	int connectionFlags = NEW_CONNECTION | CACHED_CONNECTION | SESSION_LIFESPAN;
 	PGconn *connection = NULL;
-	NodeConnectionKey nodeConnectionKey;
-	NodeConnectionEntry *nodeConnectionEntry = NULL;
-	bool entryFound = false;
-	bool needNewConnection = true;
-	char *userName = CurrentUserName();
+	MultiConnection *mconnection =
+		GetNodeConnection(connectionFlags, nodeName, nodePort);
```

Comment:
Are we in a position to remove the call to PurgeConnection from ReportRemoteError now?